src/libclipper/include/clipper/redis.hpp

```diff
@@ -234,6 +234,14 @@ bool add_application(redox::Redox& redis, const std::string& appname,
 bool delete_application(redox::Redox& redis, const std::string& appname);
 
 /**
+ * Lists the names of all applications registered with Clipper.
+ *
+ * \return Returns a vector of application names as strings. If no
+ * applications were found, an empty vector will be returned.
+ */
+std::vector<std::string> list_application_names(redox::Redox& redis);
```

Comment:
@giulio-zhou Also wrote a version of this function in #123 (https://github.com/ucbrise/clipper/pull/123/files#diff-e6b5389d668a7df7cf397284a349b35a). We should make sure to dedupe them and use whichever PR gets merged first.